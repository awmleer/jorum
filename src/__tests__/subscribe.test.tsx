import * as TestRenderer from 'react-test-renderer'
import * as React from 'react'
import {bloc, Consumer, Provider, Subscribe} from '..'
import {BehaviorSubject} from 'rxjs'


@bloc
class BehaviorSubjectBloc {
  data$ = new BehaviorSubject('this is initial value')
}

it('subscribe component', function () {
  const renderer = TestRenderer.create(
    <Provider of={BehaviorSubjectBloc}>
      <Consumer of={BehaviorSubjectBloc}>
        {(bloc) => (
          <Subscribe to={bloc.data$}>
            {data => (
              <div>
                {data}
              </div>
            )}
          </Subscribe>
        )}
      </Consumer>
    </Provider>
  )
  expect(renderer.toJSON()).toMatchSnapshot()
})
