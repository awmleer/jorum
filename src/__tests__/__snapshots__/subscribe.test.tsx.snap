// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subscribe component 1`] = `
<div>
  this is initial value
</div>
`;
